cmd/internal/cli/pull.go

```diff
@@ -304,6 +316,61 @@ func pullRun(cmd *cobra.Command, args []string) {
 		if err != nil {
 			sylog.Fatalf("%v\n", err)
 		}
+	case OrasProtocol:
```

Comment:
Would it be possible to move that code to a sub-function? I think it would keep the code more readable by limiting the length of the switch statement. It would also ease testing since we could have a unit test to cover some of the error cases for that function.
(Ultimately, it might be good to do the same for the library case)

Reply:
Agree with @gvallee here. @ikaneshiro suggested we handle this in a follow-up PR, which I think is fine. Before we merge this PR let's make some issues tracking this as tech-debt so we make sure we follow up